manifests/system_probe.pp

```diff
@@ -0,0 +1,53 @@
+include datadog_agent::params
+class datadog_agent::system_probe(
+  Boolean $enabled = false,
+  Optional[String] $log_file = undef,
+  Optional[String] $sysprobe_socket = undef,
+
+  Boolean $service_enable = true,
+  String $service_ensure = 'running',
+  Optional[String] $service_provider = undef,
+) inherits datadog_agent::params {
+
+  if $::operatingsystem == 'Windows' {
+    # Datadog does not currently support Windows and macOS platforms for Network Performance Monitoring
+    return()
```

Comment:
```suggestion
    fail('Network performance monitoring is only supported on Linux.')
```
An alternative would be to print something and then return, which is less aggressive than `fail()`. In any case, I think we should notify the user instead of continuing silently.